modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/pages/apps/ListApps.es.js

```diff
@@ -34,13 +34,51 @@ import {
 	STATUSES,
 } from './constants.es';
 
-export const Actions = () => {
+export const Actions = (
+	checkRequiredFields,
+	getDataDefinition,
+	getFormViews,
+	setCurrentApp,
+	setDeployCallback,
+	setMissingFieldsModalVisible
+) => {
 	const {getStandaloneURL} = useContext(AppContext);
 	const {deployApp, undeployApp} = useDeployApp();
 
+	const isFormViewMissingFields = (app) => {
+		return getDataDefinition(app.dataDefinitionId).then(
+			(dataDefinition) => {
+				setCurrentApp(app);
+
+				getFormViews(
+					dataDefinition.id,
+					dataDefinition.defaultLanguageId
+				).then((formViews) => {
+					formViews = checkRequiredFields(formViews, dataDefinition);
+
+					return formViews.find(({id}) => id === app.dataLayoutId)
+						.missingRequiredFields;
+				});
+			}
+		);
+	};
+
```

Comment:
@naraandrade is there possibility move this function creation to out of this component/function and avoid passing many new props to Actions, and passing only one instead of many. What do you think? 

Example:

Create a new function to pass as prop to the Actions:
```javascript
function validateFormViewMissingRequiredFields(app) {
        if(isFormViewMissingFields(app)) {
                return new Promise((resolve, reject) => {
			setDeployCallback({reject, resolve});
			setMissingFieldsModalVisible(true);
		});
	}
}

export const Actions = (validateFormViewMissingRequiredFields) => { ... }
```

and inside action function:

```javascript
{
     action: (app) => {
          return app.active ? undeployApp(app) : (validateFormViewMissingRequiredFields(app) ?? deployApp(app));
     }
}
```

Reply:
👏

Reply:
Yes, I'll do it :) Thank you